### ISSUE TITLE
[feat] additionalMutableHooks support for `react-x/immutability`

### ISSUE THREAD

**Issue by christopher-buss** [Rel1cx/eslint-react#1924]
### Problem Description

From https://github.com/Rel1cx/eslint-react/issues/1904

```tsx
export function useWorld(): World {
  return useContext(WorldContext)!;
}

function Component() {
  const world = useWorld(); // returns a external mutable store

  const onClose = useCallback(() => {
    world.set( { ... });
  }, [world]);

  return <AnotherComponent onClose={onClose} />;
}
```

Reported twice, correctly per the current contract: `world` is captured from component scope, `set` mutates, and the handler escapes render. But `world` is an external store exactly like a `navigate` object, where writing to it is intended.

`additionalRefHooks` would silence it if we registered `useWorld` there, but the world isn't a ref, so that just hides a misdeclaration in config. I'd propose a `additionalMutableHooks` covering the navigation-hook list would fit.

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users
